compiler/rustc_middle/src/hir/map/mod.rs

```diff
@@ -364,7 +364,11 @@ impl<'hir> Map<'hir> {
         match node.node {
             OwnerNode::ImplItem(impl_item) => Some(&impl_item.generics),
             OwnerNode::TraitItem(trait_item) => Some(&trait_item.generics),
-            OwnerNode::Item(Item {
+            OwnerNode::ForeignItem(ForeignItem {
+                kind: ForeignItemKind::Fn(_, _, generics),
+                ..
+            })
```

Comment:
Could this potentially break existing (invalid) code? Likely worth it to land anyways, but might need a crater run to gauge the fallout.

Reply:
There is an inconsistency here between `hir().get_generics` and the custom implementation in `generics_of`.  I am tempted to correct it by adding more cases to `get_generics` and use it in `generics_of`.  I don't see how it could break code, but I'll need to review the calls before giving an definite answer.